snuba/migrations/snuba_migrations/transactions/0012_transactions_add_spans.py

```diff
@@ -0,0 +1,63 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, Float, Nested, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
```

Comment:
You will also have to add a Bloomfilter index to the `spans.op` column and to the `spans.group` column.

See
https://github.com/getsentry/snuba/blob/ae6aaae1b815273945a2c6c73c427921f87320d7/snuba/migrations/snuba_migrations/metrics/templates.py#L133-L140

Reply:
According to the [documentation](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#available-types-of-indices), a bloom filter index is only used with the functions `equals/notEquals/in/notIn/has`.

If I'm understanding this correctly, this means that we can use it for a search like

```sql
has(spans.op, 'db')
```

However, if we want a search for multiple span op/group pairs like `[('db', 1234), ('http.client', 5678)]`, would we'd have to use `AND`/`OR`s like

```sql
(has(spans.op, 'db') OR has(spans.op, 'http.client')) AND (has(spans.group, 1234) OR has(spans.group, 5678))
```

in order to leverage the bloom filter index right?

Reply:
Not exactly. 
You would use the bloomfilter to filter down the transactions row before the arrayJoin. 
So 
```
(has(spans.op, 'db') OR has(spans.op ....)) AND (has(spans.group ....))
```
Then, on the result of the arrayjoin you would run the equality as you would do today with measurements or span_ops to narrow down the results to the exact rows you want out of the array join.

The bloom filter would operate early in the pipeline narrowing down the blocks to scan, then within the blocks you are scanning, if you do an arrayjoin, you sill have to pick only the rows you care for.